include/fmt/chrono.h

```diff
@@ -391,12 +393,62 @@ struct formatter<std::chrono::duration<Rep, Period>, Char> {
     void on_fill(Char fill) { f.spec.fill_ = fill; }
     void on_align(alignment align) { f.spec.align_ = align; }
     void on_width(unsigned width) { f.spec.width_ = width; }
+    void on_precision(unsigned precision) { f.pspec.precision = precision; }
+    void end_precision() {}
 
     template <typename Id> void on_dynamic_width(Id arg_id) {
       f.width_ref = make_arg_ref(arg_id);
     }
+
+    template <typename Id> void on_dynamic_precision(Id arg_id) {
+      f.precision_ref = make_arg_ref(arg_id);
+    }
   };
 
+  template <typename Ch, typename SpecHandler>
+  static const Ch* parse_precision(const Ch* begin, const Ch*,
+                                   SpecHandler&& handler, std::false_type) {
+    if (*begin == '.') {
+      handler.on_error("precision not allowed for this argument type");
+    }
+    return begin;
+  }
+
+  template <typename Ch, typename SpecHandler>
+  static const Ch* parse_precision(const Ch* begin, const Ch* end,
+                                   SpecHandler&& handler, std::true_type) {
+    if (*begin == '.') {
+      begin = internal::parse_precision(begin, end, handler);
+    }
+    return begin;
+  }
+
+  template <typename FormatContext, typename OutputIt>
+  OutputIt format_value(FormatContext& fctx, OutputIt out, Rep val) {
+    basic_format_specs<Char> specs;
+    if (pspec.has_precision()) {
+      specs.precision = pspec.precision;
+      specs.type = 'f';
+    }
+    typedef typename format_context_t<OutputIt, Char>::type context;
+    context ctx(out, {}, {}, fctx.locale());
+    arg_formatter<output_range<OutputIt, Char>> af(ctx, &specs);
+    internal::custom_formatter<Char, context> cf(ctx);
+    const auto arg = internal::make_arg<context>(val);
+    if (!visit_format_arg(cf, arg))
+      visit_format_arg(af, arg);
+    return ctx.out();
```

Comment:
I think this can be significantly simplified by using `format_to(out, "{:.{}f}", val, precision)` or something like that.

Reply:
Sure, but with my performance-hat on I tried to avoid reparsing.